code/modules/projectiles/gun.dm

```diff
@@ -151,9 +151,14 @@
 		return ..() //Pistolwhippin'
 
 /obj/item/weapon/gun/proc/Fire(atom/target, mob/living/user, clickparams, pointblank=0, reflex=0)
+
+
 	if(!user || !target) return
 
 	add_fingerprint(user)
+	if((user.client.prefs.toggles_secondary & SAFETY_CHECK) && user.a_intent != I_HURT) //Check this first to save time.

```

Comment:
I realized, it is also possible that user doesn't have a client. Poke a `user.client` into this if chain here.

So: `if (user.client && (user.client.prefs.toggles_secondary & SAFETY_CHECK) && user.a_intent != I_HURT)`

Reply:
Resolved.